test/utils/TestUniversalResolver.js

```diff
@@ -560,6 +560,45 @@ contract('UniversalResolver', function (accounts) {
       )
       expect(addrRet).to.equal(dummyOffchainResolver.address)
     })
+    it('should propagate HttpError', async () => {
+      const urWithHttpErrorAbi = new ethers.Contract(
+        universalResolver.address,
+        [
+          ...universalResolver.interface.fragments,
+          'error HttpError((uint16,string)[])',
+        ],
+        ethers.provider,
+      )
+      const errorData = urWithHttpErrorAbi.interface.encodeErrorResult(
+        'HttpError',
+        [[[404, 'Not Found']]],
+      )
+      const extraData = ethers.utils.defaultAbiCoder.encode(
+        ['bool', 'address', 'string[]', 'bytes', '(bytes4,bytes)[]'],
+        [
+          false,
+          dummyOffchainResolver.address,
+          ['http://universal-offchain-resolver.local/'],
+          '0x',
+          [[resolveCallbackSig, errorData]],
+        ],
+      )
+      const responses = batchGateway.encodeFunctionResult('query', [
+        [true],
+        [errorData],
+      ])
+
+      try {
+        await urWithHttpErrorAbi.callStatic.resolveSingleCallback(
+          responses,
+          extraData,
+        )
+        expect(false).to.be.true
```

Comment:
`expect(false).to.be.true` this seems weird - is this supposed to be like this?

Reply:
yes, we're using waffle chai matchers which are pretty outdated so afaik this is the only way to assert that a call has reverted if i want to access the data of the error

Reply:
there were actually a few missing so i added those as well